toolchain/driver/BUILD

```diff
@@ -13,10 +15,66 @@ filegroup(
     data = glob(["testdata/**/*.carbon"]),
 )
 
+# Build rules to construct the install data tree for the Carbon toolchain.
+#
+# The directory `install` should mirror a Unix-y [FHS] style installation of
+# Carbon under some prefix (`/usr`, `/usr/local`, or possibly a home directory
+# or a custom prefix). When installed, user facing binaries, which we largely
+# expect to be solely the `carbon` driver will be at `install/bin/carbon`. We
+# collect data, private executables, and other internals used by the toolchain
+# in `install/lib/carbon`. While using a `lib` subdirectory isn't a perfect fit
+# for the FHS, having a single location where private data is collected is
+# significantly superior to spreading them across the system. This also matches
+# similar patterns used by Clang itself and several other language toolchains
+# and standard libraries.
```

Comment:
This comment is seeming kind of long. Perhaps some of this is more appropriate for the PR description? Do we need all of this argument about the best approach going forward?